### PR TITLE
chore: deprecate `scale` argument of `centr_eigen()` and `centr_eigen_tmax()`

### LINKED ISSUE #1531: Deprecate scale argument of `centr_eigen()` and `centr_eigen_tmax()`
https://github.com/igraph/rigraph/issues/1353#issuecomment-2371834047 by @szhorvat 

https://github.com/igraph/igraph/pull/2674

### PR DESCRIPTION
**Pull request by maelle**
Fix #1531

Also added tests as there were none.

### CODE REVIEW

File: tests/testthat/_snaps/centralization.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      i The function always behaves as if `scale` were TRUE. The argument will be removed in the future.
+
+# centr_degree_tmax() deprecated argument
+
+    Code
+      c <- centr_degree_tmax(g)
+    Condition
+      Warning:
+      The `loops` argument of `centr_degree_tmax()` must be explicit as of igraph 2.0.0.
+      i Default value (`FALSE`) will be dropped in next release, add an explicit value for the loops argument.
```
Comment (maelle) [inline tests/testthat/_snaps/centralization.md:26]: I actually wonder why an explicit value is needed here.

Comment (szhorvat) [reply tests/testthat/_snaps/centralization.md:26]: There is no reasonable default. The user should be forced to make a decision. The decision can only be made based on an understanding of where the network came from—it can't be deduced by looking at the network structure.

File: R/centralization.R [hunk 2]
```diff
@@ -510,4 +531,23 @@ centr_eigen <- centralization_eigenvector_centrality_impl
 #'   `/`(centr_eigen_tmax(g))
 #' centr_eigen(g, normalized = TRUE)$centralization
 #' @cdocs igraph_centralization_eigenvector_centrality_tmax
-centr_eigen_tmax <- centralization_eigenvector_centrality_tmax_impl
+centr_eigen_tmax <- function(graph = NULL,
+                             nodes = 0,
+                             directed = FALSE,
+                               scale = deprecated()) {
```
Comment (szhorvat) [inline R/centralization.R:537]: Inconsistent indentation.

File: R/centralization.R [hunk 3]
```diff
@@ -481,7 +481,28 @@ centr_clo_tmax <- centralization_closeness_tmax_impl
...
+      The argument will be removed in the future."
+    )
+  }
+
+  centralization_eigenvector_centrality_impl(
+    graph = graph,
+    directed = directed,
+    options = options,
+    normalized = normalized
+  )
```
Comment (szhorvat) [inline R/centralization.R:504]: I would be explicit and pass `scale=TRUE` explicitly here instead of leaving it to the interface generator to provide this default.

Passing this is necessary to prevent a deprecation warning coming form the C core.

File: R/centralization.R [hunk 4]
```diff
@@ -510,4 +531,23 @@ centr_eigen <- centralization_eigenvector_centrality_impl
...
+      details = "The function always behaves as if `scale` were TRUE.
+      The argument will be removed in the future."
+    )
+  }
+
+  centralization_eigenvector_centrality_tmax_impl(
+    graph = graph,
+    nodes = nodes,
+    directed = directed
+  )
```
Comment (szhorvat) [inline R/centralization.R:552]: Same comment as before: I recommend passing `scale=T` explicitly.

**Comment by szhorvat** [comment]
Other than the comments, LGTM.

File: tests/testthat/_snaps/centralization.md (hunk 1 above)
Comment (maelle) [reply tests/testthat/_snaps/centralization.md:26]: Ooooh I understand now. I think the current message is slightly misleading: I thought support for TRUE was getting dropped, not support for not choosing. :sweat_smile:

Comment (maelle) [reply tests/testthat/_snaps/centralization.md:26]: But now that I know what it means I am not sure how to rephrase the message.

Comment (szhorvat) [reply tests/testthat/_snaps/centralization.md:26]: Hm, to me it seems clear (?), it just needs some articles: "The default values", "in the next release". Is it the mention of what the default was (i.e. `FALSE`) that you find confusing?

**Comment by maelle** [comment]
probably ready for a merge, minus the exact version in deprecation messages maybe

File: R/centralization.R [hunk 5]
```diff
@@ -288,7 +288,7 @@ centr_degree_tmax <- function(graph = NULL,
     lifecycle::deprecate_warn(
       when = "2.0.0",
       what = "centr_degree_tmax(loops = 'must be explicit')",
-      details = "Default value (`FALSE`) will be dropped in next release, add an explicit value for the loops argument."
+      details = "The default value (currently `FALSE`) will be dropped in the next release, add an explicit value for the loops argument."
```
Comment (szhorvat) [inline R/centralization.R:291]:
```suggestion
      details = "The default value (currently `FALSE`) will be dropped in the next release. Add an explicit value for the loops argument."
```

Maybe split the sentence?

**Review by szhorvat — APPROVED** [review]

File: R/centralization.R [hunk 6]
```diff
@@ -288,7 +288,7 @@ centr_degree_tmax <- function(graph = NULL,
     lifecycle::deprecate_warn(
       when = "2.0.0",
       what = "centr_degree_tmax(loops = 'must be explicit')",
-      details = "Default value (`FALSE`) will be dropped in next release, add an explicit value for the loops argument."
+      details = "The default value (currently `FALSE`) will be dropped in the next release. Add an explicit value for the loops argument."
```
Comment (krlmlr) [inline R/centralization.R:291]:
```suggestion
      details = "The default value (currently `FALSE`) will be dropped in the next release. Add an explicit value for the `loops` argument."
```

File: R/centralization.R [hunk 7]
```diff
@@ -481,7 +481,29 @@ centr_clo_tmax <- centralization_closeness_tmax_impl
...
+                        directed = FALSE,
+                        scale = deprecated(),
+                        options = arpack_defaults(),
+                        normalized = TRUE) {
+
+  if (lifecycle::is_present(scale)) {
+    lifecycle::deprecate_soft(
+      "2.1.3",
+      "centr_eigen(scale = )",
+      details = "The function always behaves as if `scale` were TRUE.
```
Comment (krlmlr) [inline R/centralization.R:494]:
```suggestion
      "2.2.0",
      "centr_eigen(scale = )",
      details = "The function always behaves as if `scale = TRUE`.
```

File: R/centralization.R [hunk 8]
```diff
@@ -510,4 +532,24 @@ centr_eigen <- centralization_eigenvector_centrality_impl
...
+centr_eigen_tmax <- function(graph = NULL,
+                             nodes = 0,
+                             directed = FALSE,
+                             scale = deprecated()) {
+
+  if (lifecycle::is_present(scale)) {
+    lifecycle::deprecate_soft(
+      "2.1.3",
+      "centr_eigen_tmax(scale = )",
+      details = "The function always behaves as if `scale` were TRUE.
```
Comment (krlmlr) [inline R/centralization.R:544]:
```suggestion
      "2.2.0",
      "centr_eigen_tmax(scale = )",
      details = "The function always behaves as if `scale = TRUE`.
```

**Review by krlmlr — APPROVED** [review]
Thanks!

**Comment by krlmlr** [comment]
Thanks!
